### ISSUE TITLE
require magit-gerrit doesn’t work if installed from MELPA

### ISSUE THREAD

**Issue by gergelypolonkai** [emacsorphanage/magit-gerrit#41]
After installing with MELPA, magit-gerrit isn’t loaded by default. Also, if I put `(require 'magit-gerrit)` in my `init.el`, I get an error:

```
Warning (initialization): An error occurred while loading `/home/epolger/.emacs.d/init.el':

File error: Cannot open load file, no such file or directory, magit-gerrit
```

The correct command seems to be

```
(add-hook 'after-init-hook (lambda () (require 'magit-gerrit)))
```

However, if I do this, pressing `R` doesn’t do anything gerrit-related in `magit-mode` (it wants to rename a branch instead).

**Comment by terranpro** [comment]
Hi,

For the second issue, I wonder if magit-gerrit couldn't detect a gerrit repo in your repository?

There's a function called `magit-gerrit-check-enable` that only enables itself in gerrit git repos.
It looks for either one of two things:
1) ssh url w/gerrit port 29418
2) `magit-gerrit-ssh-creds` is set and matches the remote url

Example, if you set `magit-gerrit-ssh-creds` to  "user@server.org" and the git repo remote url is "ssh://user@server.org:29418/my/project" it should detect and enable gerrit

But if the remote url were, "https://github.com/another-user/awesome-repo" then neither case would match, and gerrit would not be enabled.

And if it's not enabled the 'R' hotkey would not be overridden and the default git rename functionality would be present

Could you check/share your remote url and `magit-gerrit-ssh-creds` settings?

Thanks

**Comment by gergelypolonkai** [comment]
I have a remote named `gerrit`, with the URL `ssh://<my-username>@<internal-gerrit-host>:29418/demand` .

When I run `M-x magit-status`, I get this message:

    Detected magit-gerrit-ssh-creds=<my-username>@<internal-gerrit-host>

So I guess `magit-gerrit-ssh-creds` detects my credentials just OK.

On the other hand, I would suggest some other checks. There is a `git-review` script which interacts with Gerrit; it checks for either a remote with the name `gerrit`, **or** a file named `gitreview` with some review board settings. Maybe `magit-gerrit` should also check these?

**Comment by terranpro** [comment]
Interesting, could certainly add those checks as well; I haven't used git-review since magit-gerrit but I'm guessing it must create that file - as I don't see it in my gerrit repos.

So it does detect and enable gerrit, does the magit status screen show gerrit reviews? if so, does the hotkey still not work?

**Comment by gergelypolonkai** [comment]
No, it needs a remote with the name `gerrit` **or** a file called `.gitreview` (I misspelled the name in my last comment; note the leading dot). Honestly, I have no idea what this `.gitreview` file should contain, I simply go with the specially named remote.

I have found some strange (undocumented?) behaviour. `R` will try to do a branch rename. However, if I move my point to a line that contains an unpushed commit, the git-review popup will… erm… pop up :)
